$OpenBSD: patch-lib_wibox_layout_fixed_lua_in,v 1.1 2013/10/07 14:28:52 dcoppa Exp $

commit 8792d6a4ba3577de79d5cfb0ef738b64424e4552
Author: Bjorn Astrom <bjoast@gmail.com>
Date:   Thu Oct 3 07:11:09 2013 +0200

wibox.layout.fixed: Fix fill space

The last widget always took up the remaining
space even though fill_space(false)
had been called on the layout.

This got broken in commit 9d333113dd4.

Signed-off-by: Uli Schlachter <psychon@znc.in>

--- lib/wibox/layout/fixed.lua.in.orig	Mon Apr  1 13:44:46 2013
+++ lib/wibox/layout/fixed.lua.in	Mon Oct  7 14:44:34 2013
@@ -27,7 +27,7 @@ function fixed:draw(wibox, cr, width, height)
         if self.dir == "y" then
             x, y = 0, pos
             w, h = width, height - pos
-            if k ~= #self.widgets or not self.fill_space then
+            if k ~= #self.widgets or not self._fill_space then
                 _, h = v:fit(w, h);
             end
             pos = pos + h
@@ -35,7 +35,7 @@ function fixed:draw(wibox, cr, width, height)
         else
             x, y = pos, 0
             w, h = width - pos, height
-            if k ~= #self.widgets or not self.fill_space then
+            if k ~= #self.widgets or not self._fill_space then
                 w, _ = v:fit(w, h);
             end
             pos = pos + w
